### ISSUE TITLE
Port twisted.internet._newtls to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6140]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6140|
|Type|enhancement|
|Created|2012-10-16 18:29:16Z|
|Branch|https://github.com/twisted/twisted/tree/newtls-py3-6140|

`twisted.internet._newtls` should run on Python 3.

<details><summary>Searchable metadata</summary>

```
trac-id__6140 6140
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_newtls_py3_6140 branches/newtls-py3-6140
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1350412156000000 1350412156000000
changetime__1354649491000000 1354649491000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

This depends on [#6172](https://github.com/twisted/twisted/issues/6172) (specifically `twisted.internet.test.test_newtls` requires `twisted.internet.test.test_tls`).

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst commented|
|-|-|

(In [36360]) Branching to 'newtls-py3-6140'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst set owner to @exarkun|
|-|-|

Again, trivial port; last SSL ticket I think.

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/newtls-py3-6140

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @itamarst|
|-|-|

Great, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst set status to `closed`|
|-|-|

(In [36520]) Merge newtls-py3-6140.

Author: itamar
Review: exarkun
Fixes: [#6140](https://github.com/twisted/twisted/issues/6140)

Mark _newtls as ported to Python 3.
